src/modules/hub.bicep

```diff
@@ -131,6 +145,207 @@ resource uploadSettingsJson 'Microsoft.Resources/deploymentScripts@2020-10-01' =
+    storageAccount
+    keyVault
+  ]
+  params: {
+    keyVaultName: keyVaultName
+    storageAccountName: storageAccountName
+    secretName: storageAccountName
+  }
+}
+
+// Azure Data Factory and Pipelines
+module dataFactory 'Microsoft.DataFactory/factories/deploy.bicep' = {
+  name: 'dataFactory'
+  params: {
+    name: dataFactoryName
+    location: location
+    tags: resourceTags
+    systemAssignedIdentity: true
+  }
+}
+
+module linkedService_keyVault 'linkedservices/linkedservice.bicep' = {
+  name: 'linkedService_keyVault'
+  dependsOn: [
+    dataFactory
+    keyVaultSecret
+  ]
+  params: {
+    dataFactoryName: dataFactoryName
+    linkedServiceName: keyVaultName
+    linkedServiceType: 'AzureKeyVault'
+    linkedServiceTypeProperties:  {
+      baseUrl: keyVaultUri
+    }
+  }
+}
+
+module linkedService_storageAccount 'linkedservices/linkedservice.bicep' = {
+  name: 'linkedService_storageAccount'
+  dependsOn: [
+    dataFactory
+    linkedService_keyVault
+  ]
+  params: {
+    dataFactoryName: dataFactoryName
+    linkedServiceName: storageAccountName
+    linkedServiceType: 'AzureBlobFS'
+    linkedServiceTypeProperties: {
+      url: reference('Microsoft.Storage/storageAccounts/${storageAccountName}', '2019-04-01').primaryEndpoints.dfs
+      accountKey: {
+        type: 'AzureKeyVaultSecret'
+        store: {
+          referenceName: keyVaultName
+          type: 'LinkedServiceReference'
+        }
+        secretName: storageAccountName
+      }
+    }
+  }
+}
+
+module msexports 'datasets/adlsv2.bicep' = {
+  name: 'msexports'
+  dependsOn: [
+    linkedService_storageAccount
+    linkedService_keyVault
+  ]
+  params: {
+    dataFactoryName: dataFactoryName
+    linkedServiceName: storageAccountName
+    compressionCodec: 'none'
+    datasetType: 'DelimitedText'
+    containerName: exportContainer.name
+  }
+}
+
+module ingestion_csv 'datasets/adlsv2.bicep' = {
+  name: 'ingestion_csv'
+  dependsOn: [
+    linkedService_storageAccount
+    linkedService_keyVault
+  ]
+  params: {
+    dataFactoryName: dataFactoryName
+    linkedServiceName: storageAccountName
+    compressionCodec: 'gzip'
+    datasetType: 'DelimitedText'
+    containerName: ingestionContainer.name
+  }
+}
+
+module ingestion_parquet 'datasets/adlsv2.bicep' = {
+  name: 'ingestion_parquet'
+  dependsOn: [
+    linkedService_storageAccount
+    linkedService_keyVault
+  ]
+  params: {
+    dataFactoryName: dataFactoryName
+    linkedServiceName: storageAccountName
+    compressionCodec: 'gzip'
+    datasetType: 'Parquet'
+    containerName: ingestionContainer.name
+  }
+}
+
+module transform_parquet 'pipelines/msexports_transform.bicep' = {
+  name: 'msexports_transform_parquet'
+  dependsOn: [
+    msexports
+    ingestion_parquet
+  ]
+  params: {
+    dataFactoryName: dataFactoryName
+    sinkDataset:  replace('${ingestionContainer.name}_Parquet', '-', '_')
+    sourceDataset:  replace('${exportContainer.name}_DelimitedText', '-', '_')
```

Comment:
nit: Should we put the source before the sink? I know it doesn't matter. Just thinking about it in logical order. Maybe the same for export container before ingestion? (Same for transform_csv.)

Reply:
Done